src/nwb_datajoint/spikesorting/spikesorting_curation.py

```diff
@@ -1,127 +1,171 @@
-        found = True
-        while found:
-            id = 'C_' + str(uuid.uuid4())[:8]
-            if len((Curation & {'curation_id': id}).fetch()) == 0:
-                found = False
+        # generate a unique number for this curation
+        id = (Curation & sorting_key).fetch('curation_id')
+        if len(id) > 0:
+            id.sort()
+            curation_id = id[-1] + 1
+        else:
+            curation_id = 0
+
+        # convert unit_ids in labels to integers for labels from sortingview.
+        new_labels = {int(unit_id): labels[unit_id] for unit_id in labels}
 
-        sorting_key['curation_id'] = id
+        sorting_key['curation_id'] = curation_id
         sorting_key['parent_curation_id'] = parent_curation_id
         sorting_key['description'] = description
-        sorting_key['labels'] = labels
+        sorting_key['curation_labels'] = new_labels
         sorting_key['merge_groups'] = merge_groups
-        sorting_key['metrics'] = metrics
+        sorting_key['quality_metrics'] = metrics
+        sorting_key['time_of_creation'] = int(time.time())
 
         Curation.insert1(sorting_key)
 
-        return sorting_key['curation_id']
+        # get the primary key for this curation
+        c_key = Curation.fetch("KEY")[0]
+        curation_key = {item: sorting_key[item] for item in c_key}
+
+        return curation_key
 
     @staticmethod
-    def get_recording_extractor(key):
+    def get_recording_extractor(key: dict):
         """Returns the recording extractor for the recording related to this curation
 
-        :param key: key to a single Curation entry
-        :type key: dict
+        Parameters
+        ----------
+        key : dict
+            SpikeSortingRecording key
+
+        Returns
+        -------
+        recording_extractor : spike interface recording extractor
 
-        returns
-            recording_extractor: spike interface recording extractor
         """
         recording_path = (SpikeSortingRecording & key).fetch1('recording_path')
         return si.load_extractor(recording_path)
 
     @staticmethod
-    def get_curated_sorting_extractor(key):
-        """Returns the sorting extractor related to this curation, with merges applied
+    def get_curated_sorting_extractor(key: dict):
+        """Returns the sorting extractor related to this curation,
+        with merges applied.
 
-        :param key: key to a single Curation entry
-        :type key: dict
+        Parameters
+        ----------
+        key : dict
+            Curation key
+
+        Returns
+        -------
+        sorting_extractor: spike interface sorting extractor
 
-        returns
-            sorting_extractor: spike interface sorting extractor
         """
         sorting_path = (SpikeSorting & key).fetch1('sorting_path')
         sorting = si.load_extractor(sorting_path)
         merge_groups = (Curation & key).fetch1('merge_groups')
         # TODO: write code to get merged sorting extractor
```

Comment:
should this call `MergedSortingExtractor` recursively in case there were multiple rounds of curations?

Reply:
I'm pretty sure this is correct. The merges should be accumulated in the merge_groups list, so a single sorting extractor can be created. We haven't tested the merge accumulation yet, but if that code doesn't work it should be easy to fix.